### PR TITLE
Add ability to update metadata royalties

### PR DESCRIPTION
**Pull request by ChewingGlass**

### CODE REVIEW

File: programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs [hunk 1]
```diff
@@ -9,12 +12,30 @@ pub struct UpdateEntryMintMetadataCtx<'info> {
...
     certificate_mint_metadata: UncheckedAccount<'info>,
     token_metadata_program: UncheckedAccount<'info>,
 }
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct Creator {
+    pub address: Pubkey,
+    pub verified: bool,
+    // In percentages, NOT basis points ;) Watch out!
+    pub share: u8,
+}
```
Comment (jpbogle) [inline programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs:27]: why not use the actual metaplex creator struct here?

File: programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs [hunk 2]
```diff
@@ -9,12 +12,30 @@ pub struct UpdateEntryMintMetadataCtx<'info> {
...
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct UpdateMetadataArgs {
```
Comment (jpbogle) [inline programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs:30]:
```suggestion
pub struct UpdateMetadataIx {
```
just convention we have in the other instruction handlers :)

File: programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs [hunk 3]
```diff
@@ -9,12 +12,30 @@ pub struct UpdateEntryMintMetadataCtx<'info> {
...
+    /// Royalty basis points that goes to creators in secondary sales (0-10000)
+    pub seller_fee_basis_points: u16,
+    /// Array of creators, optional
+    pub creators: Option<Vec<Creator>>,
+    pub primary_sale_happened: Option<bool>,
+}
 
-pub fn handler(ctx: Context<UpdateEntryMintMetadataCtx>) -> ProgramResult {
+pub fn handler(ctx: Context<UpdateEntryMintMetadataCtx>, args: UpdateMetadataArgs) -> ProgramResult {
```
Comment (jpbogle) [inline programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs:38]:
```suggestion
pub fn handler(ctx: Context<UpdateEntryMintMetadataCtx>, ix: UpdateEntryMintMetadataIx) -> ProgramResult {
```

File: programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs (hunk 1 above)
Comment (ChewingGlass) [reply programs/cardinal-namespaces/src/instructions/update_entry_mint_metadata.rs:27]: Anchor is not smart enough to pick it up and put it in the IDL

**Comment by jpbogle** [comment]
LGTM thanks for the cleanup!

**Comment by jpbogle** [comment]
Will be merging this and adding a few integration tests around it

**Pull request merged**